consensus-types/blocks/factory.go

```diff
@@ -288,3 +288,21 @@ func BuildSignedBeaconBlockFromExecutionPayload(
 
 	return NewSignedBeaconBlock(fullBlock)
 }
+
+// BeaconBlockContainerToSignedBeaconBlock converts BeaconBlockContainer (API response) to a SignedBeaconBlock.
+// This is particularly useful for using the values from API calls.
+func BeaconBlockContainerToSignedBeaconBlock(obj *eth.BeaconBlockContainer) (interfaces.SignedBeaconBlock, error) {
+	if obj.GetPhase0Block() != nil {
+		return NewSignedBeaconBlock(obj.GetPhase0Block())
+	}
+	if obj.GetAltairBlock() != nil {
+		return NewSignedBeaconBlock(obj.GetAltairBlock())
+	}
+	if obj.GetBellatrixBlock() != nil {
+		return NewSignedBeaconBlock(obj.GetBellatrixBlock())
+	}
+	if obj.GetBlindedBellatrixBlock() != nil {
+		return NewSignedBeaconBlock(obj.GetBlindedBellatrixBlock())
+	}
+	return nil, errors.New("container has no block")
```

Comment:
How about a switch statement here?

```go
switch obj.Block.(type) {
    case *eth.SignedBeaconBlock:
       return NewSignedBeaconBlock(obj.GetPhase0Block())
    ...
}
```

Reply:
this was copied from somewhere else. presumably we use a series of if statements because the overhead is lower than switches. I agree that a switch would make the logical grouping of things easier to see in general, but the bulk of this function would be that one switch so I think the benefit of clarity would be pretty small? Come to think of it, it probably would be best to reverse order these tests from newest fork to oldest, given that processing new blocks is more time critical than old blocks. But it probably doesn't matter much in practice.

Reply:
actually on second thought a type switch is probably more efficient than all these calls and individual type assertions, and it's more readable, so I'll make that change :+1:

Reply:
changed to switch, ptal